vowpalwabbit/search.cc

```diff
@@ -298,6 +300,59 @@ struct search_private
                   // the "real" decision step but this really only matters for caching purposes
   v_array<v_array<action_cache>*>
       memo_foreach_action;  // when foreach_action is on, we need to cache TRAIN trajectory actions for LEARN
+
+  ~search_private(){
+    if(all)
+    {
+      delete truth_string;
+      delete pred_string;
+      delete bad_string_stream;
+      neighbor_features.delete_v();
+      timesteps.delete_v();
+      if (cb_learner)
+        learn_losses.cb.costs.delete_v();
+      else
+        learn_losses.cs.costs.delete_v();
+      if (cb_learner)
+        gte_label.cb.costs.delete_v();
+      else
+        gte_label.cs.costs.delete_v();
+
+      condition_on_actions.delete_v();
+      learn_allowed_actions.delete_v();
+      ldf_test_label.costs.delete_v();
+      active_uncertainty.delete_v();
+      for (size_t i = 0; i < active_known.size(); i++) active_known[i].delete_v();
+      active_known.delete_v();
+
+      if (cb_learner)
+        allowed_actions_cache->cb.costs.delete_v();
+      else
+        allowed_actions_cache->cs.costs.delete_v();
+
+      train_trajectory.delete_v();
+
+      for (auto& ar : ptag_to_action) delete ar.repr;
```

Comment:
Add this to ~action_repr()?

Reply:
Going to hold off on this because action_repr is pushed around by value, so we'd need to workout proper ownership as well when we do this